### DISCUSSION TITLE
Kwil and TSN metrics

### DISCUSSION THREAD

**Discussion by zolotokrylin** [trufnetwork/node#133] [Q&A]
@brennanjl @KwilLuke for marketing purposes we would like to display TSN real metrics. What are the current capabilities of Kwil to provide network metrics (what are the metrics, what's the API)?

**Comment by brennanjl** [comment]
Unfortunately, Kwil doesn't really have any metrics information, however it is clear how one could be built. Similar to other blockchain networks, Kwil nodes doesn't actually index metrics information on the node; this is a non-trivial task, and usually requires a separate server to serve as a block explorer.

Since Kwil is built on CometBFT, a lot of this data (such as validators, txs, block info, etc.) can be accessed via the CometBFT RPC. This is configured to localhost:26657 by default, but can be changed: https://docs.kwil.com/docs/daemon/config/settings/#chain.

To actually query the RPC, you can simply use curl. We have a few examples in our Wiki: https://github.com/kwilteam/kwil-db/wiki/Node-Recovery#using-cometbft-inspect.

Our options seems to be:
1. Use the existing CometBFT RPC querying capabilities to get any metrics possible. This is likely pretty rudimentary. The API can be seen here: https://docs.cometbft.com/v0.38/rpc/
2. Build a basic tool on top of the CometBFT RPC to index important data.

I will also ask around the CometBFT community to see if there are any known tools that perform indexing on CometBFT data.

**Reply by brennanjl** [reply]
Spoke with Informal Systems (the main team behind Comet); it seems this is an active project in the community right now: https://github.com/cometbft/cometbft/pull/658/files

**Reply by zolotokrylin** [reply]
@brennanjl thank you for your effort!
Let's continue in this Goal https://github.com/orgs/truflation/projects/2/views/1?pane=issue&itemId=56924068
